Gems/Atom/RPI/Code/Source/RPI.Public/Material/Material.cpp

```diff
@@ -67,34 +67,39 @@ namespace AZ
             ShaderReloadNotificationBus::MultiHandler::BusDisconnect();
             if (!m_materialAsset.IsReady())
             {
-                AZ_Error(s_debugTraceName, false, "Material::Init failed because shader asset is not ready. materialAsset uuid=%s",
+                AZ_Error(
+                    s_debugTraceName,
+                    false,
+                    "Material::Init failed because shader asset is not ready. materialAsset uuid=%s",
                     materialAsset.GetId().ToFixedString().c_str());
                 return RHI::ResultCode::Fail;
             }
 
             if (!m_materialAsset->InitializeNonSerializedData())
             {
-                AZ_Error(s_debugTraceName, false, "Material::InitializeNonSerializedData is not supposed to fail. materialAsset uuid=%s",
+                AZ_Error(
+                    s_debugTraceName,
+                    false,
+                    "Material::InitializeNonSerializedData is not supposed to fail. materialAsset uuid=%s",
                     materialAsset.GetId().ToFixedString().c_str());
                 return RHI::ResultCode::Fail;
             }
 
-            // Cache off pointers to some key data structures from the material type...
+            auto instanceHandler = MaterialInstanceHandlerInterface::Get();
+
+            if (instanceHandler)
+            {
+                m_instanceData = instanceHandler->RegisterMaterialInstance(this);
+            }
+            else
+            {
+                return RHI::ResultCode::Fail;
```

Comment:
Please add an assert or error here, so we can see why the `Init` failed.

Reply:
added an assert